shell/browser/electron_browser_context.cc

```diff
@@ -151,6 +151,45 @@ ElectronBrowserContext::ElectronBrowserContext(const std::string& partition,
 #endif
 }
 
+ElectronBrowserContext::ElectronBrowserContext(base::FilePath partition,
```

Comment:
Copied and pasted overloaded constructors are always source of problems. Someone will forget modify one of them when a change is needed and it will break. Best practice is to call one constructor from the other one.

The constructor above(the original one), can call this constructor first, then call 
` base::PathService::Get(DIR_SESSION_DATA, &path_);
  if (!in_memory && !partition.empty())
    path_ = path_.Append(FILE_PATH_LITERAL("Partitions"))
                .Append(base::FilePath::FromUTF8Unsafe(
                    MakePartitionName(partition)));`

Would that work?

Reply:
I agree with this observation, the reason I added a new ElectronBrowserContext for accepting a FilePath input was because it seemed to be the norm with other API which accepts absolute paths as input eg: session.loadExtension() accepts a FilePath as input. I guess the electron reviewers will need to decide if the current approach is fine.

One way we could keep a single constructor for ElectronBrowserContext is by adding FilePath as a parameter as well. Again, I am not sure if this is a preferred approach.

If we are ok with it, we could pass the patch for review to the main electron feature branch and gather their responses.

Reply:
Just adding an addtional observation about the proposed changes, this is indeed possible, though the check would also need to cater for the absolute path and non-absolute path cases.  

  
```
if (!in_memory && !partition.empty()){
    FilePath temp_path(partition)

    if(temp_path.IsAbsolute()) // fromPath()
        path_= std::move(temp_path)
    else // fromPartition
        path_ = path_.Append(FILE_PATH_LITERAL("Partitions"))
                   .Append(base::FilePath::FromUTF8Unsafe(
                       MakePartitionName(partition)));
}

```